src/audio/copier/copier.c

```diff
@@ -1659,8 +1659,13 @@ static uint64_t copier_get_processed_data(struct comp_dev *dev, uint32_t stream_
 	uint64_t ret = 0;
 
 	if (cd->endpoint_num && cd->bsource_buffer != input) {
-		if (stream_no < cd->endpoint_num)
-			ret = comp_get_total_data_processed(cd->endpoint[stream_no], 0, input);
+		if (stream_no < cd->endpoint_num) {
+			if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw)
+				ret = input == dev->direction ? cd->hd->total_data_processed : 0;
```

Comment:
Secondly, the expression
```
bool source = dev->direction == SOF_IPC_STREAM_CAPTURE;
```
does make sense - `dev->direction` can indeed have the value of `SOF_IPC_STREAM_CAPTURE` and the result of that comparison by definition has a boolean type, so that's a perfectly valid expression, no need to "fix" there

Reply:
Thanks, changed!